### PR TITLE
snap7: init at 1.4.2

### PR DESCRIPTION
**Pull request by freezeboy**
###### Motivation for this change

Make snap7 library package

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by prusnak — CHANGES_REQUESTED** [review]
`meta` section is missing; see https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes for more info

File: pkgs/development/libraries/snap7/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "snap7";
+  version = "1.4.0";
```
Comment (symphorien) [inline pkgs/development/libraries/snap7/default.nix:5]: there is a 1.4.2 version available when clicking "download" on the homepage. (on sourceforge)

Comment (freezeboy) [reply pkgs/development/libraries/snap7/default.nix:5]: My bad, I will change it

Comment (freezeboy) [reply pkgs/development/libraries/snap7/default.nix]: fixed

**Comment by freezeboy** [comment]
@symphorien is it ok for you now ?

File: pkgs/development/libraries/snap7/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    url = "mirror://sourceforge/snap7/${version}/snap7-full-${version}.7z";
+    sha256 = "1n5gs8bwb6g9vfllf3x12r5yzqzapmlq1bmc6hl854b8vkg30y8c";
+    postFetch = ''
+      ${p7zip}/bin/7z x $downloadedFile
+      mkdir $out
+      cp -r snap7-full-${version}/* $out/
+    '';
+  };
+
+  outputs = [ "out" "doc" ];
```
Comment (symphorien) [inline pkgs/development/libraries/snap7/default.nix:17]:
```suggestion
  outputs = [ "out" "dev" "doc" ];
```

File: pkgs/development/libraries/snap7/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  makefile = "x86_64_linux.mk";
+  makeFlags = [ "LibInstall=$(out)/lib" ];
+
+  preBuild = "cd build/unix";
+  preInstall = ''
+    mkdir -p $out/lib
+    mkdir -p $out/include
```
Comment (symphorien) [inline pkgs/development/libraries/snap7/default.nix:25]:
```suggestion
    mkdir -p $dev/include
```

**Comment by symphorien** [comment]
The new dev output is not correctly populated:
```
/nix/store/4xazmlc1sk2q8mjfqnqcsslkinhw907i-snap7-1.4.2-dev
├── include
│   └── include
└── nix-support
    └── propagated-build-inputs
```

**Comment by freezeboy** [comment]
is it ok to link the lib folder from $out to $dev ?

**Comment by freezeboy** [comment]
oh you meant the include folder maybe ? I fixed that locally

**Comment by symphorien** [comment]
no you should move it. The goal of multiple outputs is that executables depending on the .so (the $out output) do not bring the headers with them at all (the $dev output).

**Comment by freezeboy** [comment]
yes so the lib folder in $out and the include folder in $dev, right ?

result
└── lib
    └── libsnap7.so
result-dev
├── include
│   └── snap7.h
└── nix-support
    └── propagated-build-inputs
result-doc
└── share
    └── doc
        └── Snap7-refman.pdf

**Review by symphorien — APPROVED** [review]
Looks good to me. You probably should squash your commits according to CONTRIBUTING.md

**Comment by Lassulus** [comment]
the commit message still says 1.4.0, should be 1.4.2

**Pull request merged**